.github/workflows/check-docs-pr-preview.yml

```diff
@@ -0,0 +1,63 @@
+# This workflow checks that PRs modifying website docs include a preview link in the PR description.
+# It enforces KubeStellar's documentation PR best practices.
+
+name: Check Docs PR Preview Link
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+    paths:
+      - 'docs/content/**'
+      - 'docs/README.md'
+      - 'docs/mkdocs.yml'
+      - 'docs/overrides/**'
+      - 'docs/scripts/**'
+      - 'docs/requirements.txt'
+      - 'docs/main.py'
+
+jobs:
+  check-preview-link:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - name: Check for Preview Link in PR Description (bash)
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          REPO: ${{ github.repository }}
+          BRANCH: ${{ github.head_ref }}
+        run: |
+          set -e
+          BODY=$(gh pr view "$PR_NUMBER" --json body -q '.body')
+          DOC_BRANCH_REGEX='^doc-'
+          OWNER=$(gh pr view "$PR_NUMBER" --json headRepositoryOwner -q '.headRepositoryOwner.login')
+          REPO_NAME=$(gh pr view "$PR_NUMBER" --json headRepository -q '.headRepository.name')
```

Comment:
FYI: these work correctly, but they seem a bit indirect. Can this information be extracted from one of the worklow's contexts?